public/app/features/alerting/unified/components/receivers/TemplateForm.tsx

```diff
@@ -1,46 +1,81 @@
 import { css } from '@emotion/css';
 import { Location } from 'history';
-import React from 'react';
-import { useForm, Validate } from 'react-hook-form';
+import React, { ReactElement, useCallback, useEffect, useState } from 'react';
+import { FormProvider, useForm, useFormContext, Validate } from 'react-hook-form';
 import { useLocation } from 'react-router-dom';
+import { useToggle } from 'react-use';
 import AutoSizer from 'react-virtualized-auto-sizer';
 
 import { GrafanaTheme2 } from '@grafana/data';
 import { Stack } from '@grafana/experimental';
-import { Alert, Button, Field, FieldSet, Input, LinkButton, useStyles2 } from '@grafana/ui';
+import {
+  Alert,
+  Button,
+  Field,
+  FieldSet,
+  Icon,
+  Input,
+  LinkButton,
+  Spinner,
+  Tab,
+  TabsBar,
+  useStyles2,
+} from '@grafana/ui';
 import { useCleanup } from 'app/core/hooks/useCleanup';
 import { AlertManagerCortexConfig } from 'app/plugins/datasource/alertmanager/types';
 import { useDispatch } from 'app/types';
 
+import {
+  AlertField,
+  TemplatePreviewErrors,
+  TemplatePreviewResponse,
+  TemplatePreviewResult,
+  usePreviewTemplateMutation,
+} from '../../api/templateApi';
 import { useUnifiedAlertingSelector } from '../../hooks/useUnifiedAlertingSelector';
 import { updateAlertManagerConfigAction } from '../../state/actions';
+import { GRAFANA_RULES_SOURCE_NAME } from '../../utils/datasource';
 import { makeAMLink } from '../../utils/misc';
 import { initialAsyncRequestState } from '../../utils/redux';
 import { ensureDefine } from '../../utils/templates';
 import { ProvisionedResource, ProvisioningAlert } from '../Provisioning';
 
+import { PayloadEditor } from './PayloadEditor';
 import { TemplateDataDocs } from './TemplateDataDocs';
 import { TemplateEditor } from './TemplateEditor';
 import { snippets } from './editor/templateDataSuggestions';
 
-interface Values {
+export interface TemplateFormValues {
   name: string;
   content: string;
 }
 
-const defaults: Values = Object.freeze({
+export const defaults: TemplateFormValues = Object.freeze({
   name: '',
   content: '',
 });
 
 interface Props {
-  existing?: Values;
+  existing?: TemplateFormValues;
   config: AlertManagerCortexConfig;
   alertManagerSourceName: string;
   provenance?: string;
 }
 export const isDuplicating = (location: Location) => location.pathname.endsWith('/duplicate');
 
+export const DEFAULT_PAYLOAD = `[
+  {
+    "annotations": {
+      "summary": "Instance instance1 has been down for more than 5 minutes"
+    },
+    "labels": {
+      "instance": "instance1"
+    },
+    "startsAt": "2023-04-01T00:00:00Z",
```

Comment:
We should maybe make these dynamic? :) 

Perhaps `startAt: now - 10m` and `endsAt: now`?